VortexEngine/src/Patterns/Multi/ZigzagPattern.cpp

```diff
@@ -159,7 +142,7 @@ void ZigzagPattern::Snake::drawSnake()
       col.adjustBrightness(m_fadeAmount * segment);
     }
     // lookup the target in the step positions array and turn it on with given color/brightness
-    Leds::setIndex(ledStepPositions[segment_position], col);
+    Leds::setIndex((LedPos)segment_position, col);
```

Comment:
This seems wrong/weird, segment_position was a counter which was used to index an array of LedPos variables, if segment_position is now an LedPos then change it's type, don't cast

Reply:
I was thinking this as well, but in lines 134-137 segment_position is used in math with segment (another uint8_t) so I feel like I'd need to change both but it looks like segment is better off as a uint8_t in other places so I wasn't sure and just wrote it like this instead. I can definitely change one or both, let me know what you think!

Reply:
you're right this is fair, in this device the zigzag steps happen to line up with the linear incrementing counter, so it really is a counter and used for math, but the translation table is just unnecessary.